pkg/controller/deployment/deployment_controller.go

```diff
@@ -417,6 +420,13 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		dc.enqueueDeployment(&d)
 		return nil
 	}
+	if !dc.podStoreSynced() {
+		// Sleep so we give the pod reflector goroutine a chance to run.
+		time.Sleep(PodStoreSyncedPollPeriod)
```

Comment:
> And where is the deployment store checked?

We don't need check deployment store in fact. We need check pod store and rs store because if DeploymentController begin to work when pod/rcs store has ***NOT***  synced, it will make incorrect scale decision. See more details how controller framwork works : https://github.com/kubernetes/kubernetes/issues/20259